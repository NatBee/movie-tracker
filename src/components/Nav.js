import React, { Component } from 'react';
import { connect } from 'react-redux';
import { logOutUser } from '../actions/index';
import { Link, NavLink } from 'react-router-dom';

class Nav extends Component {

  logOutUser = (user) => {
    this.props.logOutUser(user);
  }

  render() {
    return (
      <div>
        <nav> 
          <NavLink to='/' className='nav'>All Movies</NavLink>
          <NavLink to='/favorites' className='nav'>Favorites</NavLink>
          <NavLink to='/login' className='nav'>Login</NavLink>
          <button onClick={this.logOutUser}>Logout</button>
        </nav>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  logOutUser: (user) => dispatch(logOutUser(user))
})

export default connect(null, mapDispatchToProps)(Nav);