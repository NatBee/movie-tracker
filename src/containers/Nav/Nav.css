.nav-menu {
  background-color: #122227;
  display: flex;
  padding: 10px 10px;
  justify-content: space-around;
  min-height: 20px;
  position: relative;
  font-family: 'Roboto', sans-serif;
}

.nav {
  color: white;
  text-decoration: none;
  background: black;
}

button {
  border: none;
  text-decoration: none;
  color: white;
  background: none;
  font-size: 16px;
  font-family: 'Roboto', sans-serif;
}