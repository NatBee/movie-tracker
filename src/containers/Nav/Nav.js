import React, { Component } from 'react';
import { connect } from 'react-redux';
import { logOutUser } from '../../actions/index';
import { NavLink, withRouter } from 'react-router-dom';
import './Nav.css';
import PropTypes from 'prop-types';

export class Nav extends Component {

  logOutUser = (user) => {
    this.props.logOutUser(user);
    this.props.history.push('/')
  }

  render() {

    if (this.props.user.name) {
     return (
        <div>
          <nav className='nav-menu'> 
            <NavLink to='/' className='nav'>All Movies</NavLink>
            <NavLink to='/favorites' className='nav'>Favorites</NavLink>
            <button className='logout-button' onClick={this.logOutUser}>Logout</button>
          </nav>
        </div>
        )
      }

    return (
      <div>
        <nav className='nav-menu'> 
          <NavLink to='/' className='nav'>All Movies</NavLink>
          <NavLink to='/favorites' className='nav'>Favorites</NavLink>
          <NavLink to='/login' className='nav'>Login</NavLink>
        </nav>
      </div>
    )
  }
}

export const mapStateToProps = (store) => ({
  user: store.current_user
})

export const mapDispatchToProps = (dispatch) => ({
  logOutUser: (user) => dispatch(logOutUser(user))
})

Nav.propTypes = {
  history: PropTypes.object,
  user: PropTypes.object,
  logOutUser: PropTypes.func,
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Nav));