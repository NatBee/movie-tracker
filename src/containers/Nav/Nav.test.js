/* eslint-disable */

import React from 'react';
import { shallow } from 'enzyme';
import Nav from './Nav';

describe('Nav', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Nav />)
  })

  it.skip('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
});