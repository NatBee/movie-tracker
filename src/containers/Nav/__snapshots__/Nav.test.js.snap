// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    user={
      Object {
        "name": "Natalie",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="nav-menu"
      >
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav"
          to="/"
        >
          All Movies
        </NavLink>
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav"
          to="/favorites"
        >
          Favorites
        </NavLink>
        <button
          className="logout-button"
          onClick={[Function]}
        >
          Logout
        </button>
      </nav>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav"
            to="/"
          >
            All Movies
          </NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav"
            to="/favorites"
          >
            Favorites
          </NavLink>,
          <button
            className="logout-button"
            onClick={[Function]}
          >
            Logout
          </button>,
        ],
        "className": "nav-menu",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "All Movies",
            "className": "nav",
            "to": "/",
          },
          "ref": null,
          "rendered": "All Movies",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Favorites",
            "className": "nav",
            "to": "/favorites",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Logout",
            "className": "logout-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Logout",
          "type": "button",
        },
      ],
      "type": "nav",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="nav-menu"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav"
            to="/"
          >
            All Movies
          </NavLink>
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav"
            to="/favorites"
          >
            Favorites
          </NavLink>
          <button
            className="logout-button"
            onClick={[Function]}
          >
            Logout
          </button>
        </nav>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/"
            >
              All Movies
            </NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav"
              to="/favorites"
            >
              Favorites
            </NavLink>,
            <button
              className="logout-button"
              onClick={[Function]}
            >
              Logout
            </button>,
          ],
          "className": "nav-menu",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "All Movies",
              "className": "nav",
              "to": "/",
            },
            "ref": null,
            "rendered": "All Movies",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Favorites",
              "className": "nav",
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Logout",
              "className": "logout-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Logout",
            "type": "button",
          },
        ],
        "type": "nav",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
